### ISSUE TITLE
Status bar

### ISSUE THREAD

**Issue by kubaraczkowski** [Rockhopper-Technologies/enlighten#20]
Hi,

Great job, this works really well!

One additional idea I have is to be able to add a "generic" status bar line to the output.
What could be a good to do that using the current code?

The `Manager` can be used to directly `write()` to the stream at a given vertical position, but since it's a bypass of the normal update system the output gets overwritten immediately.

A `Status` class would be an idea, however the manager is also expecting a single type of `counter_class` so this won't mix nicely with the default behavior.

Thanks!

**Comment by kubaraczkowski** [comment]
Except the "obvious" way of abusing the Counter class and providing/updating the `desc` property, with format `"{desc}{fill}"`

**Comment by avylove** [comment]
So you're looking for something you can just print to the screen?
Have you tried `Counter.format()`? If you set `enable=False` to the counter, it won't display the normal progress bar as well.


    import time
    import enlighten

    manager = enlighten.get_manager()
    ticks = manager.counter(total=100, desc='Ticks', unit='ticks', enabled=0)

    for num in range(100):
        time.sleep(0.1)
        ticks.update()
        if not num % 5:
            print(ticks.format())

Or would you need it to have a different format from `bar_format`?

**Comment by kubaraczkowski** [comment]
Thanks for your example!

Well, indeed, perhaps just using the highly flexible formatting system and slightly abusing the Counter class is all that's needed!

My example below, adding and updating a 'status bar' in addition to a handful of counters.
```python
    status_bar_text = f"ver: {__version__}"
    status_bar = self.enlighten_manager.counter(
            desc=status_bar_text, counter_format="{desc}{fill}"
    )
    
    time.sleep(5)
    status_bar.desc = "Hello world"
    status_bar.refresh()  
```

**Comment by avylove** [comment]
Oh, are you trying to keep something at the bottom of the screen to let users know some dynamic information? Yeah, you could abuse it that way. I'd have to think if making that easier makes sense or not. I can see how it could.

**Comment by kubaraczkowski** [comment]
Right, exactly.
In fact I am running sth that spits loads of log information (wow, thanks,
the logging integration is great) so I wanted to give also a dynamic text
indication of the general status, or, hm, let’s say IP address of the
connected server or so.

Making it _much_ easier might be too much of a change, perhaps an example
would be enough?

**Comment by avylove** [comment]
I think some status bar class makes sense. In the meantime, do you have what you need to workaround it for now?

**Comment by kubaraczkowski** [comment]
Yes, the workaround was fine, thank you.

I looked at the code on how to implement a StatusBar class, but saw it was
not immediately obvious without modifying also pieces of the manager, etc.
and then decided to contact you.

**Comment by avylove** [comment]
This has been implemented in [1.6.0](https://github.com/Rockhopper-Technologies/enlighten/releases/tag/1.6.0). See the [documentation](https://python-enlighten.readthedocs.io/en/stable/examples.html#status-bars) for more information.

I'm going to go ahead and close this, but please provide feedback here once you try it out.

**Comment by kubaraczkowski** [comment]
Awasome! Great
